spec/factories/measure_factory.rb

```diff
@@ -116,7 +116,19 @@
           :base_regulation,
           base_regulation_id: measure.measure_generating_regulation_id,
           base_regulation_role: measure.measure_generating_regulation_role,
-          effective_end_date: evaluator.base_regulation_effective_end_date || Time.zone.today.in(10.years),
+          effective_end_date: evaluator.base_regulation_effective_end_date,
+        )
+      end
+    end
+
+    trait :with_unapproved_base_regulation do
```

Comment:
I'd push to replace the `after(:create)` with the already existing `generating_regulation` transient property because its a lot simpler

```ruby
generating_regulation { create :base_regulation, :unapproved }
```

Advantages are 
* the regulation gets created before the measure which depends upon it
* keeps logic creating regulations in the regulation factory
* the regulation is responsible for determining its own ids
* the measure just reuses the ids from the regulation
* you can ignore dates

Reply:
Amazing. Thanks Jeremy